### PR TITLE
Add back the `--payment-method` option for `solidus:install`

### PR DESCRIPTION
**Pull request by elia**
## Summary

- Another batch of installer cleanups and fixes
- More installer tests
- Adds support for PayPal and Bolt when using the _classic_ frontend

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [x] I have added automated tests to cover my changes.
~- [ ] I have attached screenshots to demo visual changes.~
~- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~
~- [ ] I have updated the readme to account for my changes.~

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Hey, @elia 👋 I think we decided to put it on hold until solidus_starter_frontend works with the payment extensions, right?

**Comment by elia** [comment]
@waiting-for-dev I though we still had an urge to restore support for PayPal as soon as possible.
Anyway this PR includes a bail out from payment methods if the frontend is not `solidus_frontend`, that should be easy to lift once we restore support for payment methods on SSF.

That works for you?

File: core/lib/generators/solidus/install/app_templates/payment_method/paypal.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+unless Bundler.locked_gems.dependencies['solidus_frontend']
+  say_status :warning, "Support for frontends other than `solidus_frontend` by `solidus_paypal_commerce_platform` is still in progress.", :yellow
+end
+
+bundle_command 'add solidus_paypal_commerce_platform'
```
Comment (waiting-for-dev) [inline core/lib/generators/solidus/install/app_templates/payment_method/paypal.rb:5]: We should skip this line if the gem is already bundled in the same vein that you're considering on `detect_payment_method_to_install` below.

File: core/lib/generators/solidus/install/app_templates/payment_method/bolt.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+unless Bundler.locked_gems.dependencies['solidus_frontend']
+  say_status :warning, "Support for frontends other than `solidus_frontend` by `solidus_bolt` is still in progress.", :yellow
+end
+
+unless Bundler.locked_gems.dependencies['solidus_auth_devise']
+  say_status :warning, "Running solidus_bolt without solidus_auth_devise is not supported.", :yellow
+end
+
+bundle_command 'add solidus_bolt'
```
Comment (waiting-for-dev) [inline core/lib/generators/solidus/install/app_templates/payment_method/bolt.rb:9]: Same here. We shouldn't run bundle if already present.

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @elia, I left some questions & suggestions, but it looks good overall. I think it'd still make more sense to have the payment extensions working with SSF first. However, you're doing a great job with the installer, so I'm perfectly fine with doing it the other way if that's going to make your work easier.

Another question to your judgment: do you think it'd make sense to remove Bolt from the solidus_frontend installer before merging this PR? That way, we wouldn't need to mess with `SKIP_SOLIDUS_BOLT` from here. As we're working with `master`, I don't think it's important if there's no option to install a payment method in the meantime.

File: core/lib/generators/solidus/install/install_generator.rb [hunk 3]
```diff
@@ -57,8 +57,8 @@ def self.exit_on_failure?
 
     def prepare_options
       @run_migrations = options[:migrate]
-      @load_seed_data = options[:seed]
-      @load_sample_data = options[:sample]
+      @load_seed_data = options[:seed] && @run_migrations
+      @load_sample_data = options[:sample] && @run_migrations
```
[the remark this answers is not in the document]
Comment (elia) [reply core/lib/generators/solidus/install/install_generator.rb]: Good call, I probably mixed up the two during some rebase 👍

File: core/lib/generators/solidus/install/install_generator.rb [hunk 4]
```diff
@@ -48,11 +48,7 @@ class InstallGenerator < Rails::Generators::AppBase
     class_option :lib_name, type: :string, hide: true, default: nil
 
     def self.source_paths
-      paths = superclass.source_paths
-      paths << File.expand_path('../templates', "../../#{__FILE__}")
-      paths << File.expand_path('../templates', "../#{__FILE__}")
-      paths << File.expand_path('templates', __dir__)
-      paths.flatten
+      ["#{__dir__}/templates"]
```
[the remark this answers is not in the document]
Comment (elia) [reply core/lib/generators/solidus/install/install_generator.rb:58]: I didn't, but did now, discovering that we can just use `source_root` as explained in the error message:

```
Could not find "config/initializers/spree.rb.tt" in any of your source paths. Please invoke Solidus::InstallGenerator.source_root(PATH) with the PATH containing your templates. Currently you have no source paths.
```

File: core/lib/generators/solidus/install/install_generator.rb [hunk 5]
```diff
@@ -130,19 +130,6 @@ def create_database
       rake 'db:create'
     end
 
-    def install_authentication
```
[the remark this answers is not in the document]
Comment (elia) [reply core/lib/generators/solidus/install/install_generator.rb]: Thanks! Removed ✂️

**Review by elia — COMMENTED** [review]
@waiting-for-dev thanks for the review! ❤️ 

> I think it'd still make more sense to have the payment extensions working with SSF first.

The way I see it we're free to release 3.3 even without SSF support in payment methods (or a partial one) and then release a patch level that just lifts the checks. But thank you for the openness, both ways will take us there 🙏 

> do you think it'd make sense to remove Bolt from the solidus_frontend installer before merging this PR?

Yes! But we probably should [release v3.2.3](https://github.com/solidusio/solidus/compare/v3.2.2...v3.2) that will let users install it (bolt) without any intervention from `solidus_frontend`. In any case I'll move ahead and remove `SKIP_SOLIDUS_BOLT` from here and open a PR against `solidus_frontend`, so we'll have everything ready 👍

**Review by waiting-for-dev — APPROVED** [review]
🚀

**Review by spaghetticode — APPROVED** [review]
Thanks @elia !

**Review by jarednorman — APPROVED** [review]

**Review by adammathys — APPROVED** [review]

**Pull request merged**